scipy/stats/_binned_statistic.py

```diff
@@ -541,55 +544,20 @@ def binned_statistic_dd(sample, values, statistic='mean',
     except TypeError:
         bins = Ndim * [bins]
 
-    # Select range for each dimension
-    # Used only if number of bins is given.
-    if range is None:
-        smin = np.atleast_1d(np.array(sample.min(axis=0), float))
-        smax = np.atleast_1d(np.array(sample.max(axis=0), float))
-    else:
-        smin = np.zeros(Ndim)
-        smax = np.zeros(Ndim)
-        for i in xrange(Ndim):
-            smin[i], smax[i] = range[i]
-
-    # Make sure the bins have a finite width.
-    for i in xrange(len(smin)):
-        if smin[i] == smax[i]:
-            smin[i] = smin[i] - .5
-            smax[i] = smax[i] + .5
-
     # Create edge arrays
-    for i in xrange(Ndim):
-        if np.isscalar(bins[i]):
-            nbin[i] = bins[i] + 2  # +2 for outlier bins
-            edges[i] = np.linspace(smin[i], smax[i], nbin[i] - 1)
-        else:
-            edges[i] = np.asarray(bins[i], float)
-            nbin[i] = len(edges[i]) + 1  # +1 for outlier bins
-        dedges[i] = np.diff(edges[i])
-
-    nbin = np.asarray(nbin)
+    if reuse is None:
+        nbin, edges, dedges = _bin_edges(sample, bins, range)
+    else:
+        edges = reuse.bin_edges
```

Comment:
I notice now that `reuse` uses the `bin_edges` and the `binnumber` elements in the named tuple but not the `statistic` the usage should be spelled out in the docstring. e.g. some users may expect to be able to leave the `statistic` argument to the `binned_statistc_dd()` call empty in which case they might get an error. Also it could be that the two disagree (e.g. named tuple uses `count` but in the function signature the `statistic=std`. Which statistic should the user expect to be returned?

Reply:
I guess this is solved.